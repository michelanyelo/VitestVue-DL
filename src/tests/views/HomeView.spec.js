import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import HomeView from "@/views/HomeView.vue";

describe("HomeViewComponent", () => {
    const wrapper = mount(HomeView)

    it("El html de HomeView se mantiene en las pruebas", () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    // fallar a proposito
    it("La etiqueta img posee una clase fixed", () => {
        const img = wrapper.find('img')

        // la etiqueta img no posee id ni class
        expect(img.classes()).toContain('img-fluid')
    })
})

