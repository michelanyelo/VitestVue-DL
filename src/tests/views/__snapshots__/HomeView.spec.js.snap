// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeViewComponent > El html de HomeView se mantiene en las pruebas 1`] = `
"<div class="home"><img alt="Vue logo" src="/src/assets/logo.png">
  <div class="posts">
    <h1>Vista de Posts</h1>
    <ul>
      <li>Post 1</li>
      <li>Post 2</li>
      <li>Post 3</li>
      <li>Post 4</li>
    </ul>
  </div>
</div>"
`;
